lib/seed_migration/migrator.rb

```diff
@@ -23,11 +23,17 @@ def up
       klass = class_from_path
       version, _ = self.class.parse_migration_filename(@path)
       raise "#{klass} has already been migrated." if SeedMigration::DataMigration.where(version: version).first
+      raise_if_missing_required_environments(klass, version)
 
       start_time = Time.now
       announce("#{klass}: migrating")
       ActiveRecord::Base.transaction do
-        klass.new.up
+        if should_apply_in_this_environment?(klass, version)
+          klass.new.up
```

Comment:
Wouldn't there also be a corresponding execution for `klass.new.down`, i.e. we only migrate down on those specific environments? We could run into a case where a down from one migration (with a specific environment) can revert migrations in other environments then?

Reply:
All the seed versions are stored in the db in the `seed_migration_data_migrations` table, even if a has seed not been applied in that env because of the `environments` restriction. It opted for it to make it easier for rollbacks. And that's exactly because of it that we don't need to develop a corresponding logic in the `down` logic.

Let me know if you agree with that.

Reply:
I think I'm fine with it, but i wonder what happens when we run into situation where for e.g. seeding a Client of name `Volkswagen` in `production`, then we do it in `:adac_production`. And we're not restricting rollback to environments, so when we just want to run a rollback `rails seed:rollback` there's a chance that it will wipe out in every environment? I guess won't be an actual issue as long as the rollback itself is different migration.

Reply:
Thanks @absessive I see your point. I will investigate and test it a bit more.

Reply:
Thanks for raising that @absessive - it was a bug, I fixed it.